test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8411,108 +8436,102 @@ def is_history_enabled(self):
             return True
         return False
 
-    def cleanup_jobs(self, extend=None, runas=None):
+    def cleanup_jobs(self):
         """
         Helper function to delete all jobs.
         By default this method will determine whether
         job_history_enable is on and will cleanup all history
         jobs. Specifying an extend parameter could override
         this behavior.
-
-        :param runas: Clean the job as
-        :type runas: str or None
         """
-        delete_xt = 'force'
+        delete_xt = 'nomailforce'
         select_xt = None
         if self.is_history_enabled():
             delete_xt += 'deletehist'
             select_xt = 'x'
-        job_ids = self.select(extend=select_xt)
+        jobs = self.status(JOB, extend=select_xt)
+        job_ids = sorted(list(set([x['id'] for x in jobs])))
+        host_pid_map = {}
+        for job in jobs:
+            if 'exec_host2' in job:
```

Comment:
While I was looking into code to confirm that first node name in exec host will MS or not, I find almost all code is looking at JOB_ATR_exec_host2 or ATTR_exechost2 so I thought it will be visible to use so I used, but if you are saying it's not visible then I will remove it.